tests/production_pause_vote_timeout.py

```diff
@@ -0,0 +1,200 @@
+    # Verify producercNode paused
+    assert paused, "producercNode still producing after finalizercNode was shutdown"
+    # Verify node0 and node1 still producing but LIB should not advance
+    assert node0.processUrllibRequest("producer", "paused", returnType=ReturnType.raw) == b'false', "node0 paused after finalizercNode was shutdown"
+    assert node1.processUrllibRequest("producer", "paused", returnType=ReturnType.raw) == b'false', "node1 paused after finalizercNode was shutdown"
+    if node0.waitForLibToAdvance(timeout=5): # LIB can advance for a few blocks first
+        assert not node0.waitForLibToAdvance(timeout=5), "LIB should not advance on node0 after finalizercNode was shutdown"
+    if node1.waitForLibToAdvance(timeout=5):
+        assert not node1.waitForLibToAdvance(timeout=5), "LIB should not advance on node1 after finalizercNode was shutdown"
+
+    Print("Restart finalizercNode")
+    finalizercNode.relaunch()
+
+    Print("Verify LIB advances after restart of finalizercNode")
+    assert node0.waitForLibToAdvance(), "node0 did not advance LIB"
+    assert node1.waitForLibToAdvance(), "node1 did not advance LIB"
+    assert producercNode.waitForLibToAdvance(), "producercNode did not advance LIB"
+
+    ####################### test 2 ######################
+
+    Print("Shutdown centerNode")
+    centerNode.kill(signal.SIGTERM)
+    assert not centerNode.verifyAlive(), "centerNode did not shutdown"
+
+    # wait some time for producercNode paused
+    paused = False
+    for i in range(0, 15):
+        time.sleep(1)
+        paused = producercNode.processUrllibRequest("producer", "paused", returnType=ReturnType.raw)
+        if paused == b'true':
+            Print(f'paused after {i} seconds after centerNode was shutdown')
+            break;
+    # Verify producercNode paused
+    assert paused, "producercNode still producing after centerNode was shutdown"
+    # Verify node0 and node1 still producing but LIB should not advance
+    assert node0.processUrllibRequest("producer", "paused", returnType=ReturnType.raw) == b'false', "node0 paused after centerNode was shutdown"
+    assert node1.processUrllibRequest("producer", "paused", returnType=ReturnType.raw) == b'false', "node1 paused after centerNode was shutdown"
+    if node0.waitForLibToAdvance(timeout=5): # LIB can advance for a few blocks first
+        assert not node0.waitForLibToAdvance(timeout=5), "LIB should not advance on node0 after centerNode was shutdown"
+    if node1.waitForLibToAdvance(timeout=5):
+        assert not node1.waitForLibToAdvance(timeout=5), "LIB should not advance on node1 after centerNode was shutdown"
+
+    Print("Restart centerNode")
+    centerNode.relaunch()
+
+    Print("Verify LIB advances after restart")
+    assert node0.waitForLibToAdvance(), "node0 did not advance LIB"
+    assert node1.waitForLibToAdvance(), "node1 did not advance LIB"
+    assert producercNode.waitForLibToAdvance(), "producercNode did not advance LIB"
+
```

Comment:
Again you need to check that `producercNode` automatically resumed block production.